### PR TITLE
Add documentation for provider and service description files

### PR DESCRIPTION
**Pull request by vigejolla**

### CODE REVIEW

File: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+grand_parent: Apps and MW
+layout: default
+---
+
+## Account Provider and Service description files
+
+A plugin which adds account providers and/or services to the system should describe them using description files, which are defined below.
+
+Each account is tied to a provider. The provider allows the user to use the account with a number of services. Each account provider can provide various types of services, and each service type can be provided by various providers.
+
```
Comment (rainemak) [inline Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md:14]: I'd add a link to the [API docs](https://sailfishos.org/develop/docs/sailfish-accounts/)  to this paragraph and write something about Providers and Services APIs can be found there. And yes, I do see that there are linking from Providers and Service types section. This way this page would have clearer focus towards explaining what Providers and Services types are.

File: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/README.md [hunk 2]
```diff
@@ -25,3 +26,7 @@ The signond component of the Accounts&SSO project stores account credentials on
 There are plans to move away from Accounts&SSO and instead implement an accounts framework specifically for Sailfish OS. It would consist of a lightweight API layer which uses private DBus connections to talk to a central accounts daemon. That accounts daemon would be responsible for servicing requests made via DBus, enforcing per-application access controls, secure storage, and OAuth2 token management and refreshing.
 
 Currently, no concrete roadmap for this work has yet been defined. Contributions are welcome.
+
+### Account plugins
+
+New account types can be added to the system via plugins. A plugin should describe the services it provides using [description files](Providers_and_Services/).
```
Comment (rainemak) [inline Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/README.md:32]: This is a bit hidden and at least we can link to Providers_and_Services as needed. Like you said, would be cool to have move levels in the navigation bar to make this kind of topics more visible.

File: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+title: Providers and Services
+permalink: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services
+parent: Accounts and SSO
+grand_parent: Apps and MW
+layout: default
+---
+
+## Account Provider and Service description files
+
+A plugin which adds account providers and/or services to the system should describe them using description files, which are defined below.
```
Comment (rainemak) [inline Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md:11]:
```suggestion
A plugin which adds account providers and/or services to the system describes them using description files, which are defined below.
```

File: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/README.md (hunk 2 above)
Comment (rainemak) [inline Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/README.md:32]:
```suggestion
New account types can be added to the system via plugins. A plugin describes the services it provides using [description files](Providers_and_Services/).
```

**Review by rainemak — APPROVED** [review]
Looks good now. Please consider https://github.com/sailfishos/docs.sailfishos.org/pull/290#discussion_r1138365248 but I'm ok to approve this already.

File: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+parent: Accounts and SSO
+grand_parent: Apps and MW
+layout: default
+---
+
+## Providers and Services
+
+Each account in the Accounts&SSO framework is tied to a provider. The provider allows the user to use the account with a number of services. Each account provider can provide various types of services, and each service type can be provided by various providers.
+
+We have a QML API for accessing account data in the framework: [Sailfish Accounts](https://sailfishos.org/develop/docs/sailfish-accounts/).
```
Comment (rainemak) [inline Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md:13]:
```suggestion
We have [Sailfish Accounts](https://sailfishos.org/develop/docs/sailfish-accounts/) QML API for accessing account data.
```
Somehow the colon and framework bugs me. Thus, above proposal.

File: Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+A plugin which adds account providers and/or services to the system describes them using description files, which are defined below.
+
+### Provider description files
+
+Each provider must have it's own provider description file. As of Sailfish OS 4.5.0 the location of these files is `/usr/share/accounts/providers` but you can check the location by running the command `pkg-config --variable=providerfilesdir libaccounts-glib` on a [build-shell](/Tools/Platform_SDK/Building_Packages/#running-arbitrary-commands).
```
Comment (rainemak) [inline Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md:19]: The link here will be affected by https://github.com/sailfishos/docs.sailfishos.org/pull/292

Might be simpler to merge https://github.com/sailfishos/docs.sailfishos.org/pull/292 first and fix the link here already. However you like.

Comment (vigejolla) [reply Reference/Core_Areas_and_APIs/Apps_and_MW/Accounts_and_SSO/Providers_and_Services/README.md]: fixed now

**Review by rainemak — APPROVED** [review]

**Pull request merged**